wom/src/main/scala/wom/values/WomFile.scala

```diff
@@ -1,67 +1,154 @@
 package wom.values
 
-import wom.types.{WomFileType, WomGlobFileType, WomSingleFileType, WomType}
+import wom.types._
 
 import scala.util.{Success, Try}
 
 sealed trait WomFile extends WomPrimitive {
+
+  // TODO: WOM: WOMFILE: This is often used as _either_ the cloud location _or_ the path within the container. Does this need to be split into two values?
   val value: String
 
   override def valueString = value
 
   // TODO: WOM: WOMFILE: When WDL supports directories this check may need more work (refactoring to subclasses?)
   override def equals(rhs: WomValue): Try[WomBoolean] = rhs match {
     case r: WomFile => Success(WomBoolean(value.equals(r.value) && womType.equals(r.womType)))
-    case r: WomString => Success(WomBoolean(value.toString.equals(r.value.toString) && womType == WomSingleFileType))
+    case r: WomString => Success(WomBoolean(value.toString.equals(r.value.toString)))
     case r: WomOptionalValue => evaluateIfDefined("==", r, equals)
     case _ => invalid(s"$value == $rhs")
   }
+
+  /**
+    * Converts the location using f() recursively converting any files referred to by this file.
+    *
+    * @param f The function to update the location.
+    * @return A new WomFile with the updated location.
+    */
+  def mapFile(f: String => String): WomFile
```

Comment:
is it full path or the filename being passed in here?

Reply:
I can't remember off the top of my head. This might even be the relative path that gets converted to a full path by the backend.

Reply:
* What does "convert" mean?  I.e. from what to what?
 * Is there any way we could use types to better define what is going in and out?